### PR TITLE
Add introductory section

### PR DESCRIPTION
**Pull request by daverodgman**
Signed-off-by: Dave Rodgman <dave.rodgman@arm.com>

### CODE REVIEW

**Review by shebuk — COMMENTED** [review]
Additional intro text looks good to me. 

Are the following lines required? I didn't understand what unified means in this context. Can the second sentence - 'Documentation is continuously improving to address gaps' or something similar.

 
This is the unified documentation for the Mbed TLS project.
This documentation is currently incomplete and will be expanded over time.

File: index.md [hunk 1]
```diff
@@ -1,5 +1,7 @@
 # Mbed TLS documentation hub
 
+Mbed TLS provides an open-source implementation of cryptographic primitives, X.509 certificate handling and the SSL/TLS and DTLS protocols. It provides a reference implementation of the [PSA Cryptography API](https://arm-software.github.io/psa-api/crypto/). The project also supports the PSA Cryptoprocessor driver interface Specification which defines an interface for cryptoprocessor drivers. The small code footprint makes the project suitable for embedded systems. It has many users, including [TF-A](https://www.trustedfirmware.org/projects/tf-a/), [TF-M](https://www.trustedfirmware.org/projects/tf-m/) and [OP-TEE](https://www.trustedfirmware.org/projects/op-tee/).
```
Comment (tom-cosgrove-arm) [inline index.md:3]: It's a bit odd to have `Specification` in title case with `driver interface` in lower case. Either `specification` or `Drive Interface`

File: index.md [hunk 2]
```diff
@@ -1,7 +1,18 @@
...
-This documentation is currently incomplete and will be expanded over time.
+Mbed TLS provides an open-source implementation of cryptographic primitives,
+X.509 certificate handling and the SSL/TLS and DTLS protocols. It provides a
+reference implementation of the [PSA Cryptography API](https://arm-software.github.io/psa-api/crypto/).
+The project also supports the PSA Cryptoprocessor Driver Interface which enables
+support for cryptoprocessor drivers. The small code footprint makes the project
+suitable for embedded systems. It has many users, including
+[TF-A](https://www.trustedfirmware.org/projects/tf-a/),
+[TF-M](https://www.trustedfirmware.org/projects/tf-m/) and
+[OP-TEE](https://www.trustedfirmware.org/projects/op-tee/).
```
Comment (tom-cosgrove-arm) [inline index.md:11]: (note, not requesting a change: once we get to line wrapping like this, my personal preference is to always start a new line when a new sentence starts. this makes later diffs much easier to read)

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

**Pull request merged**